### PR TITLE
Adding filtered activity page

### PR DESCRIPTION
**Pull request by electrified**

### CODE REVIEW

File: src/server/offenders/offender/offender.controller.ts [hunk 1]
```diff
@@ -118,9 +118,28 @@ export class OffenderController {
...
+  async getActivityFiltered(
+    @Param('crn') crn: string,
+    @Param('filter') filter: ActivityFilter,
+  ): Promise<OffenderActivityViewModel> {
+    const convictionId = await this.sentenceService.getConvictionId(crn)
+
+    return this.activityPageCommon(crn, this.activityService.constructContactFilter(filter, { convictionId }), filter)
+  }
+
+  async activityPageCommon(
```
Comment (electrified) [inline src/server/offenders/offender/offender.controller.ts:135]: Move this common logic elsewhere?

File: src/server/offenders/offender/activity/activity.service.ts [hunk 2]
```diff
@@ -212,4 +215,28 @@ export class ActivityService {
       sensitive: contact.sensitive || false,
     }
   }
+
+  constructContactFilter(filter: ActivityFilter, options: GetContactsOptions): GetContactsOptions {
+    const contactDateFrom = DateTime.now().minus({ years: 1 }).toUTC().toISO() // TODO this needs to be more sophisticated and take into consideration the date of the last breach
```
Comment (electrified) [inline src/server/offenders/offender/activity/activity.service.ts:220]: Was hoping I can rely on a reusable function elsewhere to get the breach date or 12 months whichever is more recent

Comment (axle-h) [reply src/server/offenders/offender/activity/activity.service.ts:220]: Mmm yeah breach dates on NSIs seem reliable so we can certainly reuse this, I have something like this for my current work so maybe I can address this in my PR. we would need to be careful with avoiding unnecessary api calls by eg optionally passing in nsis/breach dates as in some cases (compliance page) you will know the breach dates already from previous calls.

File: src/server/offenders/offender/offender.controller.ts (hunk 1 above)
Comment (axle-h) [reply src/server/offenders/offender/offender.controller.ts:135]: is it not just common to this controller? Seems fine here if so? I was thinking at some point splitting this controller up into the existing folders in this module, including colocating the views as appropriate. wrong time to be refactoring right now though.

**Review by axle-h — DISMISSED** [review]

**Comment by axle-h** [comment]
I can see how e2e testing of the filters might be brittle. But could at least get one test in there and be safe like eg results include expected contact & exclude another rather than results are exactly expected contacts?

File: src/server/offenders/offender/activity/activity.service.ts (hunk 2 above)
Comment (axle-h) [reply src/server/offenders/offender/activity/activity.service.ts:220]: Ive fixed this within my current work

**Review by axle-h — APPROVED** [review]

**Pull request merged**